osu.Game.Rulesets.Catch/UI/CatchPlayfield.cs

```diff
@@ -57,6 +57,17 @@ public CatchPlayfield(BeatmapDifficulty difficulty, Func<CatchHitObject, Drawabl
             };
         }
 
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            RegisterPool<Droplet, DrawableDroplet>(1);
+            RegisterPool<TinyDroplet, DrawableTinyDroplet>(1);
+            RegisterPool<Fruit, DrawableFruit>(1);
+            RegisterPool<Banana, DrawableBanana>(1);
+            RegisterPool<JuiceStream, DrawableJuiceStream>(1);
+            RegisterPool<BananaShower, DrawableBananaShower>(1);
```

Comment:
Only one of everything at the start is a little... stringent, don't you think? At least I'd allow a couple of fruits up front, and definitely a lot more droplets and bananas? Not sure what the overhead is for creating-on-the-fly when pool limit is exceeded, but it shouldn't be a large memory footprint cost to increase these a bit and eliminate that overhead entirely.

Reply:
I could have guessed numbers without any clue but I doubt setting initial capacity will make any noticeable impact. Something like this tends to be over-allocated.

Reply:
It's always going to be a guess, but in general I prefer expending a little bit of memory to try to prevent allocations during gameplay that could lead to stutters.

In my simple test, instantiating + adding a `DrawableFruit` takes 0.08ms, but this on a PC with top-end IPC and lots of multi-core headroom. Instantiating just 10 fruits which I'd say is a lower minimum, could be very expensive on lower end/mobile platforms.

Definitely want to see a little bit larger initial capacity here. Somewhere around 10-20 for the fruits and 5-10 for juicestreams, as an example.

Reply:
i think the best way is to take a heavy beatmap (insane/expert/aspire) and judge requirements. pools should aim to never be exceeded (with obvious exceptions being things like centipede).

Reply:
I tested more.
If I'm understanding correctly, the DHO instantiation time will not be noticeable as long as multiple DHOs are not instantiated in one frame.

But currently all nested DHOs are instantiated at once when the parent DHO became alive, and will likely be noticeable as a `BananaShower` can have hundreds of bananas (it currently noticeably stutters when the banana shower became alive).
I think this should be fixed to make nested drawables instantiated incrementally.
I measured some difficult maps and the maximum number of hit objects varies greatly between maps.

- `Fruit` A stream map had more than 100, but others are 30~50.
- `Droplet` 20~50.
- `TinyDroplet` typically ~20, .
- `Banana` typically ~70, one map with a long spinner had 257, but I'm pretty sure more long spinners exist in other ranked maps.

Reply:
I'd say the upper numbers you state there (and maybe 100 for banana) sound reasonable.